subxt/src/storage/utils.rs

```diff
@@ -37,3 +42,32 @@ pub(crate) fn storage_address_root_bytes<Address: StorageAddress>(addr: &Address
     write_storage_address_root_bytes(addr, &mut bytes);
     bytes
 }
+
+/// Strips the first 16 bytes (8 for the pallet hash, 8 for the entry hash) off some storage address bytes.
+pub(crate) fn strip_storage_addess_root_bytes(
+    address_bytes: &mut &[u8],
+) -> Result<(), StorageAddressError> {
+    if address_bytes.len() >= 16 {
+        *address_bytes = &address_bytes[16..];
+        Ok(())
+    } else {
+        Err(StorageAddressError::UnexpectedAddressBytes)
+    }
+}
+
+/// Strips the first few bytes off a hash to possibly skip to the plan key value,
+/// if [`StorageHasher::hash_contains_unhashed_key()`] for this StorageHasher.
+///
+/// Returns `Err(..)` if there are not enough bytes.
+/// Returns `Ok(())` otherwise
+pub fn strip_storage_hash_bytes(
```

Comment:
I think I said this before but maybe lost the comment; does this need to be `pub`? And if not, let's just not `pub use utils` at all from `subxt::storage`, and then we can just have everythign in here be `pub` rather than `pub(crate)`

Reply:
But moreover, I think this is only ever used in `storage_key.rs` si let's just have it in that file :) (I think the others are here because they are used in more than one place but would need to confirm)